### PR TITLE
Add cube

### PR DESCRIPTION
**Pull request by c0mplikati0n**
I felt it was worth merging at this point so we can work around new features. I made sure to make the function compatible with the scene info, there are of course still changes to be made to this function. Feel free to make changes if I did something inefficiently.

The cube button at the top is only temporary.

### CODE REVIEW

**Review by notalfredo — COMMENTED** [review]
One last thing. If you click on the this button. You should see an overlay of where the object is going to place. I think it will be nice if you can click where the overlay is and then it places the object there


image

File: yarn.lock [hunk 1]
```diff
@@ -240,10 +240,10 @@
   resolved "https://registry.npmjs.org/@react-spring/types/-/types-9.6.1.tgz"
   integrity sha512-POu8Mk0hIU3lRXB3bGIGe4VHIwwDsQyoD1F394OK7STTiX9w4dG3cTLljjYswkQN+hDSHRrj4O36kuVa7KPU8Q==
 
-"@react-three/drei@9.79.4":
-  version "9.79.4"
-  resolved "https://registry.npmjs.org/@react-three/drei/-/drei-9.79.4.tgz"
-  integrity sha512-Qyy2dPTLpEs3XfvLo2Nyneht0hRqIgIjel78A9MUxq3VxbCse79iCFZZ0xD8gIof33L4jfQOrGghKK988orNOA==
+"@react-three/drei@^9.80.6":
```
[the remark this answers is not in the document]
Comment (c0mplikati0n) [reply yarn.lock:243]: Ah, I think I tried updating my version of react, not really sure how this happened.

File: components/ui/tool-panel/toolPanel.tsx [hunk 2]
```diff
@@ -44,6 +49,14 @@ export function ToolPanel(objectButtonPress: props){
          Save
       </button>
 
+      <LineSeparator/>
+
+      {/* Temporary butoon to add a cube */}
+      <button className = "bg-graySubFill text-white hover:bg-blue-500 rounded-lg p-1"
```
[the remark this answers is not in the document]
Comment (c0mplikati0n) [reply components/ui/tool-panel/toolPanel.tsx:55]: I like this suggestion, I'll be sure to use that implementation instead

**Review by notalfredo — APPROVED** [review]
I fixed 2 issues 
- When cube is added its not halfway in the grid plane
- Renders group scene correctly 

Anyways good job !

**Pull request merged**